_posts/tsukori/2016-04-28-regextalk.md

```diff
@@ -0,0 +1,11 @@
+---
+layout: post
+author: tsukori
+title: "Regular Expressions Enhanced"
+---
+
+Professor Rex Dwyer from NC State came to give a talk about the re module on python and what he has done to modify it. He showed us real world application of python programming on DNA sequences which was really interesting. He showed us lookarounds which didn't really make a lot of sense to me? They seemed like they just cut strings in specific places without deleting the spot where it split it at, which was cool, but seemed like it did the same thing as the split function. 
+
+He also showed us fuzzy matching which is where it tries to find words that are close to the desired ones even if they aren't exactly what it is and shows you how many deletions and substitutions it had to do in the process. He used the function on the exact same sentence with the requirement that it had at least 1 error, so the algorithm got greedy and grabbed parts of other words to create errors. This was pretty cool and funny as well. 
```

Comment:
These are common in genetics work, where genes mutate all the time but you still need to be able to group them even if there are small differences..